pkg/kv/postgres/store.go

```diff
@@ -0,0 +1,237 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type Driver struct{}
+
+type Store struct {
+	Pool   *pgxpool.Pool
+	Params *Params
+	closed bool
+}
+
+type Entries struct {
+	rows  pgx.Rows
+	entry *kv.Entry
+	err   error
+}
+
+const (
+	DriverName = "postgres"
+
+	defaultTableName = "kv"
+	paramTableName   = "lakefskv_table"
+)
+
+//nolint:gochecknoinits
+func init() {
+	kv.Register(DriverName, &Driver{})
+}
+
+func (d *Driver) Open(ctx context.Context, name string) (kv.Store, error) {
+	// TODO(barak): should we handle Open reuse the same store based on name
+	config, err := pgxpool.ParseConfig(name)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", kv.ErrDriverConfiguration, err)
+	}
+	pool, err := pgxpool.ConnectConfig(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", kv.ErrConnectFailed, err)
+	}
+	defer func() {
+		// if we return before store uses the pool, free it
+		if pool != nil {
+			pool.Close()
+		}
+	}()
+
+	// acquire connection and make sure we reach the database
```

Comment:
At least in a test maybe this should repeat, in case the app is ready before the DB.

Reply:
At this point I'll let the app handle backoff and retry.